sources/api/pluto/src/aws.rs

```diff
@@ -0,0 +1,47 @@
+use aws_config::default_provider::credentials::DefaultCredentialsChain;
+use aws_config::imds;
+use aws_smithy_types::retry::{RetryConfig, RetryConfigBuilder};
+use aws_types::region::Region;
+use aws_types::SdkConfig;
+use snafu::{ResultExt, Snafu};
+use std::time::Duration;
+
+// Max request retry attempts
+const MAX_ATTEMPTS: u32 = 10;
+const IMDS_CONNECT_TIMEOUT: Duration = Duration::from_secs(2);
+
+#[derive(Debug, Snafu)]
+pub(super) enum Error {
+    #[snafu(display("Failed to build IMDS client: {}", source))]
+    SdkImds {
+        source: imds::client::error::BuildError,
+    },
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+pub(crate) async fn sdk_imds_client() -> Result<imds::Client> {
+    imds::Client::builder()
+        .max_attempts(MAX_ATTEMPTS)
+        .connect_timeout(IMDS_CONNECT_TIMEOUT)
+        .build()
+        .await
+        .context(SdkImdsSnafu)
+}
+
+pub(crate) fn sdk_retry_config() -> RetryConfig {
+    RetryConfigBuilder::new().max_attempts(MAX_ATTEMPTS).build()
+}
```

Comment:
Do these need to be `pub(crate)`?
```suggestion
async fn sdk_imds_client() -> Result<imds::Client> {
    imds::Client::builder()
        .max_attempts(MAX_ATTEMPTS)
        .connect_timeout(IMDS_CONNECT_TIMEOUT)
        .build()
        .await
        .context(SdkImdsSnafu)
}

fn sdk_retry_config() -> RetryConfig {
    RetryConfigBuilder::new().max_attempts(MAX_ATTEMPTS).build()
}
```